spark/src/main/scala/org/apache/hadoop/fs/versioned/VersionedFileSystem.scala

```diff
@@ -0,0 +1,138 @@
+package org.apache.hadoop.fs.versioned
+
+import java.net.URI
+import java.util.Objects
+
+import com.gu.tableversions.metastore.VersionPaths
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs._
+import org.apache.hadoop.fs.permission.FsPermission
+import org.apache.hadoop.util.Progressable
+
+class VersionedFileSystem extends FileSystem {
+
+  private var baseURI: URI = _
+  private var baseFS: FileSystem = _
+  private var mappings: Map[String, String] = _
+  private var version: String = _
+
+  override def initialize(name: URI, conf: Configuration): Unit = {
```

Comment:
It would be useful to have some logging in this method to be able to debug the way that a versioned filesystem is initialised.

Reply:
It would be nice, however with a non-cache FS, that might litter the output. We'll have to try

Reply:
Good point.